src/roles/make-copy-sshid-automatic/tasks/main.yml

```diff
@@ -0,0 +1,29 @@
+- name: Wait for ssh to be ready
+  include_role:
+    name: common
+    tasks_from: wait-for-ssh
+  vars:
+    ssh_host: "{{ hostname }}"
+    host_username: "{{ component_username }}"
+    
+- name: Copy SSH Public Key to VSD (ignoring errors)
+  authorized_key:
+      user: "{{ vsd_custom_username | default(vsd_default_username) }}"
+      key: "{{ lookup('file', lookup('env','HOME') + '/.ssh/id_rsa.pub') }}"
+  remote_user: "{{ vsd_custom_username | default(vsd_default_username) }}"
+  become: "{{ 'no' if vsd_custom_username | default(vsd_default_username) == 'root' else 'yes' }}"
+  vars:
+    ansible_become_pass: "{{ vsd_custom_password | default(vsd_default_password) }}"
+  ignore_errors: yes
+  when: component_type|default("NONE") == 'vsd'
```

Comment:
You are not passing the component_type from  the playbook. Where does it come from? Do you need to add it to copy_sshid_*.yml ?